### PR TITLE
Unicode lambda

### PR DESCRIPTION
**Pull request by NoahStoryM**
##### Checklist

- [x] Feature
- [x] documentation
- [x] test

### Description of change

Add `case-λ`~,`case-λ:`,`pcase-λ:`,`pλ:`,`λ:`,`popt-λ:`,`opt-λ:`~.

### CODE REVIEW

File: typed-racket-doc/typed-racket/scribblings/reference/legacy.scrbl
Comment (sorawee) [inline typed-racket-doc/typed-racket/scribblings/reference/legacy.scrbl:1]: I don't think we should add/change anything to legacy or compatibility libraries.

Comment (NoahStoryM) [reply typed-racket-doc/typed-racket/scribblings/reference/legacy.scrbl:1]: > I don't think we should add/change anything to legacy or compatibility libraries.

If this file is not modified, an error will be reported (I forgot to record the specific error message).

Comment (NoahStoryM) [reply typed-racket-doc/typed-racket/scribblings/reference/legacy.scrbl:1]:
```
/usr/share/racket/pkgs/typed-racket-compatibility/typed/scheme/base.rkt:20:22: module: identifier already required
  at: case-λ
  in: typed-racket/base-env/prims
  also provided by: scheme/base
  compilation context...:
   /usr/share/racket/pkgs/typed-racket-compatibility/typed/scheme/base.rkt
  location...:
   /usr/share/racket/pkgs/typed-racket-compatibility/typed/scheme/base.rkt:20:22
  context...:
   /usr/share/racket/collects/compiler/private/cm-minimal.rkt:700:0: compile-zo*
   /usr/share/racket/collects/compiler/private/cm-minimal.rkt:633:0: compile-zo*/cross-compile
   /usr/share/racket/collects/compiler/private/cm-minimal.rkt:449:15
   /usr/share/racket/collects/compiler/private/cm-minimal.rkt:435:12: build
   /usr/share/racket/collects/compiler/private/cm-minimal.rkt:406:0: maybe-compile-zo
   /usr/share/racket/collects/compiler/private/cm-minimal.rkt:210:0: compile-root
   /usr/share/racket/collects/compiler/private/cm-minimal.rkt:317:16
   /usr/share/racket/collects/compiler/private/cm-util.rkt:14:0: ormap-strict
   [repeats 4 more times]
   /usr/share/racket/collects/compiler/private/cm-minimal.rkt:210:0: compile-root
   /usr/share/racket/collects/compiler/private/cm-minimal.rkt:317:16
   /usr/share/racket/collects/compiler/private/cm-util.rkt:14:0: ormap-strict
```

Comment (NoahStoryM) [reply typed-racket-doc/typed-racket/scribblings/reference/legacy.scrbl:1]: Ah, I read the wrong file in the reply, sorry!

**Comment by sorawee** [comment]
@NoahStoryM do you still want to work on this? It appears there's an error in the latest commit.

**Comment by NoahStoryM** [comment]
> @NoahStoryM do you still want to work on this? It appears there's an error in the latest commit.

Sorry, I just saw this reply now. I think the error is because it was tested with Racket v8.11, which does not define `case-λ`.

**Comment by NoahStoryM** [comment]
@sorawee This PR has been open for a while, could you consider merging it?

**Comment by sorawee** [comment]
I think @samth would be a good reviewer for this PR.

**Pull request merged**